packages/pocket-ic/HOWTO.md

```diff
@@ -141,3 +141,227 @@ let res1 = pic.await_call(msg_id1).unwrap();
+        WasmResult::Reply(data) => {
+            let http_response: HttpResponse = decode_one(&data).unwrap();
+            assert_eq!(http_response.body, body);
+        }
+        WasmResult::Reject(msg) => panic!("Unexpected reject {}", msg),
+    };
+
+    // There should be no more pending canister http outcalls.
+    let canister_http_requests = pic.get_canister_http();
+    assert_eq!(canister_http_requests.len(), 0);
+}
+```
+
+Note that the URL of the canister HTTP outcall must either start with `https://` or target `localhost`.
+
+In the live mode (see the section "Live Mode" for more details), the canister HTTP outcalls are processed
+by actually making an HTTP request to the URL specified in the canister HTTP outcall.
+
+Here is a sketch of a test for a canister making canister HTTP outcalls in the live mode:
+
+```rust
+#[tokio::test]
+async fn test_canister_http_live() {
+    use candid::{Decode, Encode, Principal};
+    use ic_cdk::api::management_canister::http_request::HttpResponse;
+    use ic_utils::interfaces::ManagementCanister;
+
+    let mut pic = PocketIcBuilder::new()
+        .with_nns_subnet()
+        .with_application_subnet()
+        .build_async()
+        .await;
+    let endpoint = pic.make_live(None).await;
+
+    // Retrieve the first canister ID on the application subnet
+    // which will be the effective canister ID for canister creation.
+    let topology = pic.topology().await;
+    let app_subnet = topology.get_app_subnets()[0];
+    let effective_canister_id = Principal::from_slice(
+        &topology.0.get(&app_subnet).unwrap().canister_ranges[0]
+            .start
+            .canister_id,
+    );
+
+    // Create an agent for the PocketIC instance.
+    let agent = ic_agent::Agent::builder()
+        .with_url(endpoint)
+        .build()
+        .unwrap();
+    agent.fetch_root_key().await.unwrap();
+
+    // Create a canister and install the test canister wasm.
+    let ic00 = ManagementCanister::create(&agent);
+    let (canister_id,) = ic00
+        .create_canister()
+        .as_provisional_create_with_amount(None)
+        .with_effective_canister_id(effective_canister_id)
+        .call_and_wait()
+        .await
+        .unwrap();
+    let test_wasm = [...];
+    ic00.install_code(&canister_id, &test_wasm)
+        .call_and_wait()
+        .await
+        .unwrap();
+
+    // Execute an update call on the test canister making a canister HTTP outcall.
+    let arg_bytes = Encode!(&()).unwrap();
+    let res = agent
+        .update(&canister_id, "canister_http")
+        .with_arg(arg_bytes)
+        .call_and_wait()
+        .await
+        .unwrap();
+    let http_response = Decode!(&res, HttpResponse).unwrap();
+    assert_eq!(http_response.body, b"...");
+}
+```
+
+Note that it is strongly discouraged to use the live mode for testing canister HTTP outcalls
+when making update calls using the PocketIC library, e.g.,
+using `pic.update_call` or `pic.submit_call` and `pic.await_call`.
```

Comment:
This paragraph is not very clear to me. Is there an "instead, if you want to do x, do it thusly..."? 
Isn't the gist that we do not recommend mixing live mode with the "tick-basd API" and it has nothing in particular to do with outcalls? Or do the outcalls exacerbate this incompatibility?

Reply:
> Or do the outcalls exacerbate this incompatibility?

Indeed. Then you'd most likely get a failure that the message was not processed within 100 rounds since the "tick-based" API won't mock any canister http outcalls (as it can't mock them out of the blue and for the sake of determinism it can't make any actual HTTP requests to a remote URL).

> Isn't the gist that we do not recommend mixing live mode with the "tick-basd API" and it has nothing in particular to do with outcalls?

I haven't found such a recommendation in the "Live Mode" section. Let me add it there and drop it from here.

Reply:
Ah, ok! But then I would actually leave the warning here as well, tbh.

Reply:
Ah it is a the top. All good